server/trillian_log_server.go

```diff
@@ -222,7 +222,7 @@ func (t *TrillianLogServer) commitAndLog(tx storage.LogTX, op string) error {
 }
 
 func protoToLeaf(proto *trillian.LeafProto) trillian.LogLeaf {
-	return trillian.LogLeaf{Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
+	return trillian.LogLeaf{SequenceNumber: proto.LeafIndex, Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
```

Comment:
Leaf doesn't have a sequence number as it's shared with the map implementation. LogLeaf does and I thought I did copy it both ways. I'll have another look at it.